### PR TITLE
deploy airship ci: create combined inventory file

### PR DESCRIPTION
**Pull request by aagate**
deploy airship ci: create combined inventory file

- added an example hosts_combined_example.yml as a proposal to combine "deploy_osh"
  "deploy_airship" inventory files and create a
  combined inventory file that can work for both osh and airship deployment.

- updated examples/inventory/hosts.yml with hosts that are necessary
  or airship deployment.

- extend inventory hosts.yml tasks now also sets following
  host groups required for airship deployment

  'airship-openstack-control-workers' to two caasp worker nodes
  'airship-ucp-worker' to two caasp worker nodes
  'airship-openstack-compute-workers' to third worker node
  'airship-kube-system-workers' to two caasp worker nodes

- switched from 'osh-deployer' host group to 'soc-deployer' in all playbooks.

- switched from  'airship-deployer' host group to 'soc-deployer' in all playbooks.
  * since a deployer can be used for both osh and airship deployment

- switched from 'airship-workers' to 'caasp-workers' in all playbooks.
  * since they are the identical there is no need to have two different
    host groups

### CODE REVIEW

File: examples/workdir/inventory/hosts_combined_example.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
```
Comment (aagate) [inline examples/workdir/inventory/hosts_combined_example.yml]: Note: hosts_combined_example file is created for review purposes only and
  includes following proposed changes:
   * added "airship-openstack-control-workers", "airship-ucp-workers"
     and "airship-openstack-compute-workers"  host groups that are
     necessary for airship deployment.
     - for CI,
        "airship-openstack-control-workers": set hosts to first two workers
        "airship-ucp-workers": set hosts to first two workers
        "airship-openstack-compute-workers": set hosts to third worker
   * rename 'osh-deployer' to 'socok8s-deployer'
     - since deployer can be used for osh and airship deployment
   * added "all" and "airship-workers" host groups
   * Questions:
     Q1: does deploy_osh role make use of something similar to
         "airship-openstack-control-workers"  or
         "airship-openstack-compute-workers" ? Do they need to be renamed?

File: examples/workdir/inventory/hosts.yml [hunk 2]
```diff
@@ -18,3 +18,23 @@ osh-deployer:
...
     ansible_user: root
+
+# added for airship
+airship-openstack-compute-workers:
+
+airship-openstack-control-workers:
+
+airship-ucp-workers:
+
+all:
```
Comment (evrardjp) [inline examples/workdir/inventory/hosts.yml]: all is a meta group and should not be redefined.

File: examples/workdir/inventory/hosts.yml [hunk 3]
```diff
@@ -18,3 +18,23 @@ osh-deployer:
 ses_nodes:
   vars:
     ansible_user: root
+
+# added for airship
+airship-openstack-compute-workers:
+
+airship-openstack-control-workers:
+
+airship-ucp-workers:
```
Comment (evrardjp) [inline examples/workdir/inventory/hosts.yml]: Why not reusing caasp workers?

File: examples/workdir/inventory/hosts_combined_example.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+# Note: for CI, set the third worker node is
+#       as airship-openstack-compute-workers
+airship-openstack-compute-workers:
+  hosts:
+    ccpci-27-worker-2:
+      ansible_host: 10.86.3.220
+      primary: yes
+
+all:
```
Comment (evrardjp) [inline examples/workdir/inventory/hosts_combined_example.yml]: To be removed.

File: playbooks/openstack-caasp_instance.yml [hunk 5]
```diff
@@ -174,6 +174,16 @@
                 hosts: "{{ masterinventory }}"
               caasp-workers:
                 hosts: "{{ workerinventory }}"
+              # TODO(aagate): set hosts to first two workers, move to airship_configure_caasp role
```
Comment (evrardjp) [inline playbooks/openstack-caasp_instance.yml]: if these are the same, why redefining groups?

File: run.sh [hunk 6]
```diff
@@ -72,6 +72,12 @@ case "$deployment_action" in
         build_images
         deploy_osh
         ;;
+    "setup_airship")
```
Comment (evrardjp) [inline run.sh]: Great to add this, can this be in a different commit though?

File: examples/workdir/inventory/hosts_combined_example.yml (hunk 4 above)
Comment (aagate) [reply examples/workdir/inventory/hosts_combined_example.yml]: Thx! Will remove

File: examples/workdir/inventory/hosts.yml (hunk 2 above)
Comment (aagate) [reply examples/workdir/inventory/hosts.yml]: Thx! Will remove

File: examples/workdir/inventory/hosts.yml (hunk 3 above)
Comment (aagate) [reply examples/workdir/inventory/hosts.yml]: This can be a subset of the total available caasp workers.

**Comment by aagate** [comment]
> Could you clarify the group definition's purposes?

Added explanation on the host groups

> Could you clarify the group definition's purposes?

Added more info describing different control plane groups and their use

File: run.sh (hunk 6 above)
Comment (aagate) [reply run.sh]: > Great to add this, can this be in a different commit though?

To test setup_airship, I have to update the inventory hosts file and add groups that airship requires, so the two changes kind of go hand in hand. I could add this in a separate commit but wont be able to test it.

Comment (aagate) [reply run.sh]: Went ahead and created separate PR to add 'setup_airship' as per your suggestion. This PR is again getting too long and might be good to just restrict it to inventory changes.

The other PR to add setup_airship is https://github.com/SUSE-Cloud/socok8s/pull/165 I will rebase this once that PR is merged.

Comment (aagate) [reply run.sh]: Done. #165 was merged. Rebased this PR.

File: examples/workdir/inventory/hosts.yml (hunk 3 above)
Comment (evrardjp) [reply examples/workdir/inventory/hosts.yml]: LGTM then

File: playbooks/generic-clean_airship.yml [hunk 7]
```diff
@@ -15,13 +15,13 @@
...
 
-- hosts: airship-deployer
+- hosts: soc-deployer
   gather_facts: no
   tasks:
     - name: Running cleanup script
       script: "{{ playbook_dir }}/../script_library/clean-airship.sh {{ clean_action | default('') }}"
 
-- hosts: airship-workers
+- hosts: caasp-workers
```
Comment (evrardjp) [inline playbooks/generic-clean_airship.yml]: I think we should make this file not only a airship thing, but a "clean my caasp setup of all those pesky helm charts" ;) 

For a different commit though.

**Comment by aagate** [comment]
rebased

**Comment by evrardjp** [comment]
Sensu stricto, the commit message should be updated, not to list what was done, but explain why it was done.

For example, the move from osh-deployer to soc-deployer can be explained with the fact this node can serve on both, and there is no reason to duplicate this in the inventory.

**Comment by aagate** [comment]
> Sensu stricto, the commit message should be updated, not to list what was done, but explain why it was done.
> 
> For example, the move from osh-deployer to soc-deployer can be explained with the fact this node can serve on both, and there is no reason to duplicate this in the inventory.

Its been explained in the note here https://github.com/SUSE-Cloud/socok8s/pull/148#discussion_r266698688 I can add some of those points to the commit message too

**Comment by aagate** [comment]
Updated commit and PR message, added reason for making changes to each point

File: examples/workdir/inventory/hosts_combined_example.yml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+      ansible_host: 10.86.3.220
+  vars:
+    ansible_user: root
+soc-deployer:
+  hosts:
+    soc-aio:
+      ansible_host: 10.86.3.69
+  vars:
+    ansible_user: root
+ses_nodes:
```
Comment (sgrasley) [inline examples/workdir/inventory/hosts_combined_example.yml]: ses_nodes has caused errors in airship deployments. Has this been resolved ?

Comment (aagate) [reply examples/workdir/inventory/hosts_combined_example.yml]: What kind of errors, are they format related i.e  specifying one node vs specifying multiple nodes?. I do have ses_nodes (single node) in my manual deployment, which seemed to work fine. From airship CI point of view, it will use a single node ses deployment too.  I haven't yet succeeded in getting all the way through a complete aiship deployment, still in deploy fix cycle.

Comment (sgrasley) [reply examples/workdir/inventory/hosts_combined_example.yml]: Last time I ran run.sh deploy_airship with ses_nodes in my inventory it tried to update docker on ses_node and failed.  If it is still a problem, it is not something for this review. Just wanted to bring it up.

**Comment by evrardjp** [comment]
I am testing this today.

**Comment by evrardjp** [comment]
nit: I have seen a discrepency: The inventory file is still named osh where it could be deployer. But this can be improved later.

Still testing it, to see if the deploy_osh step works.
